### PR TITLE
Adding isSimulated methods to be used in simulate mapping validation work

### PR DESCRIPTION
**Pull request by masseyke**
This PR lays the groundwork for being able to validate mappings in the simulate ingest API (see #106440 for the full draft). It adds `isSimulated` methods to BulkRequest, SimulateBulkRequest, and BulkShardRequest. It also adds an `exception` field to SimulateIndexResponse that will be used to hold any mapping validation exceptions (in a future PR).

### CODE REVIEW

**Comment by masseyke** [comment]
This builds on the simulate ingest work from #101409.

File: server/src/main/java/org/elasticsearch/action/ingest/SimulateIndexResponse.java [hunk 1]
```diff
@@ -45,13 +53,15 @@ public SimulateIndexResponse(
         long version,
         BytesReference source,
         XContentType sourceXContentType,
-        List<String> pipelines
+        List<String> pipelines,
+        Exception exception
```
Comment (dakrone) [inline server/src/main/java/org/elasticsearch/action/ingest/SimulateIndexResponse.java:57]: Perhaps annotate this with `@Nullable`?

**Review by dakrone — APPROVED** [review]
LGTM, I left one comment

**Review by parkertimmins — APPROVED** [review]
Looks good to me

**Pull request merged**